framework/src/st_things/things_stack/utils/things_network.c

```diff
@@ -318,13 +333,100 @@ void things_wifi_soft_ap_sta_left(void)
 	THINGS_LOG_V(TAG, "T%d --> %s", getpid(), __FUNCTION__);
 }
 
+void things_wifi_list_cleanup()
+{
+	if (g_wifi_scan_info != NULL) {
+		// free old wifi list
+
+		access_point_info_s* piter;
+		access_point_info_s* pdel;
+
+		piter = pdel = g_wifi_scan_info;
+
+		while (piter != NULL) {
+			pdel = piter;
+			piter = piter->next;
+
+			things_free(pdel);
+		}
+
+		g_wifi_scan_info = NULL;
+	}
+}
+
 void things_wifi_scan_done(wifi_manager_scan_info_s **scan_result, int res)
 {
 	THINGS_LOG_V(TAG, "T%d --> %s", getpid(), __FUNCTION__);
 	/* Make sure you copy the scan results onto a local data structure.
 	 * It will be deleted soon eventually as you exit this function.
 	 */
-//TO DO
+	if (scan_result == NULL) {
+		return;
+	}
+
+	things_wifi_list_cleanup();
+
+	g_wifi_count = 0;
+	wifi_manager_scan_info_s *wifi_scan_iter = *scan_result;
+	access_point_info_s* pinfo = NULL;
+	access_point_info_s* p_prev_info = NULL;
+
+	while (wifi_scan_iter != NULL) {
+		THINGS_LOG_V(TAG, "WiFi AP SSID: %-20s, WiFi AP BSSID: %-20s, WiFi Rssi: %d\n",
+			   wifi_scan_iter->ssid, wifi_scan_iter->bssid, wifi_scan_iter->rssi);
+
+		if (wifi_scan_iter->ssid != NULL) {
+			pinfo = (access_point_info_s*)things_malloc(sizeof(access_point_info_s));
+
+			if (g_wifi_scan_info == NULL) {
+				g_wifi_scan_info = pinfo;
+			}
+
+			if (p_prev_info == NULL) {
+				p_prev_info = pinfo;
+			}
+
+			pinfo->next = NULL;
+			p_prev_info->next = pinfo;
+
+			snprintf(pinfo->e_ssid, MAX_SSID_LEN, "%s", wifi_scan_iter->ssid);
+			snprintf(pinfo->bss_id, MAX_SSID_LEN, "%s", wifi_scan_iter->bssid);
```

Comment:
In this case, bss_id means mac address of AP.
If the mac address has ipv6 interface, bss id's length should be 30(included null).
Please refer.

@harish-191 It might less than MAX_SSID_LEN.
Could you pelase check it.

@JeongJunSik 
In generally. mac address likes as "xx-xx-xx-xx-xx-xx" or "xxxx-xxxx-xxxx-xxxx-xxxx-xxxx".

Reply:
@JeongJunSik @Hyunjun85 fixed.